RT_UPDATE/basepop_single.R

```diff
@@ -0,0 +1,613 @@
+    
+    SRB <- tibble(SRB    = SRB,
+                  cohort = refDate_start:refDate)
+    
+  }
+  
+  # Separate births into M and F
+  Bt <- Bt %>% 
+    left_join(SRB, by = c("year" = "cohort")) %>% 
+    mutate(Bm = B * SRB  / (1 + SRB),
+           Bf = B - Bm)
+  
+  
+  
+  # --- Reverse-survival of infants -----------------------------------------
+  # Use infant mortality (Lx values) to reverse-survive births and estimate
+  # age 0–9 populations for females and males.
+  # final part
+  pop_hat_f <- mx1dt |>
+    filter(country_code == !!country_code) |>
+    as_tibble() |>
+    select(year, age, mx = mxF) |>
+    # could try to warp to PC shape here,
+    # but uncertain infants. Maybe using
+    # an a0 assumption it'd be doable.
+    # need cohorts to structure reverse survival
+    mutate(cohort = year - age - 1,
+           age_int = 1,
+           ax = ifelse(age == 0,
+                       lt_rule_1a0_ak(M0 = mx, Sex = "f"),
+                       0.5),
+           qx = lt_id_ma_q(nMx = mx, nax = ax, AgeInt = age_int)) |>
+    filter(between(cohort, refDate_start,refDate),
+           between(year, refDate_start, (refDate + 1)),
+           age < 10) |>
+    arrange(cohort, age) |>
+    group_by(cohort) |>
+    mutate(lx = lt_id_q_l(nqx = qx, radix = radix),
+           dx = lx * qx,
+           Lx = lx - dx * (1-ax)) |>
+    filter(year == max(year)) |>
+    select(age, Lx, cohort) |>
+    left_join(Bt, by = join_by(age)) |>
+    left_join(SRB, by = join_by(cohort)) |>
+    mutate(pop_hat_f = Lx * Bf) %>%
+    ungroup() %>% 
+    select("age", "pop_hat_f", "cohort")
+  
+  # for males
+  pop_hat_m <- mx1dt |>
+    filter(country_code == !!country_code) |>
+    as_tibble() |>
+    select(year, age, mx = mxM) |>
+    # could try to warp to PC shape here,
+    # but uncertain infants. Maybe using
+    # an a0 assumption it'd be doable.
+    # need cohorts to structure reverse survival
+    mutate(cohort = year - age - 1,
+           age_int = 1,
+           ax = ifelse(age == 0,
+                       lt_rule_1a0_ak(M0 = mx, Sex = "f"),
+                       0.5),
+           qx = lt_id_ma_q(nMx = mx, nax = ax, AgeInt = age_int)) |>
+    filter(between(cohort, refDate_start,refDate),
+           between(year, refDate_start, (refDate + 1)),
+           age < 10) |>
+    arrange(cohort, age) |>
+    group_by(cohort) |>
+    mutate(lx = lt_id_q_l(nqx = qx, radix = radix),
+           dx = lx * qx,
+           Lx = lx - dx * (1-ax)) |>
+    filter(year == max(year)) |>
+    select(age, Lx, cohort) |>
+    left_join(Bt, by = join_by(age)) |>
+    left_join(SRB, by = join_by(cohort)) |>
+    mutate(pop_hat_m = Lx * Bm) %>%
+    ungroup() %>% 
+    select("age", "pop_hat_m", "cohort")
+  
+  # --- Visualization -------------------------------------------------------
+  # Plots compare estimated infant population (black) to observed (red).
+  # Helpful for assessing reverse survival model fit.
+  females <- pop_hat_f |>
```

Comment:
plots are fab but they belong in the the ODAP_backend wrapper basepop(), not in the DemoTools function.